src/LibLog/LibLog.cs

```
@@ -1393,46 +1448,63 @@ internal class Log4NetLogger
+                    UnaryExpression instanceCast = Expression.Convert(instanceParam, _loggerType);
+                    ParameterExpression levelParam = Expression.Parameter(typeof(object));
+                    UnaryExpression levelCast = Expression.Convert(levelParam, _logEventLevelType);
+                    _isEnabledForDelegate = GetIsEnabledFor(_loggerType, _logEventLevelType, instanceCast, levelCast, instanceParam, levelParam);
 
-                Type loggingEventType = Type.GetType("log4net.Core.LoggingEvent, log4net");
+                    Type loggingEventType = Type.GetType("log4net.Core.LoggingEvent, log4net");
 
-                _createLoggingEvent = GetCreateLoggingEvent(instanceParam, instanceCast, levelParam, levelCast, loggingEventType);
+                    _createLoggingEvent = GetCreateLoggingEvent(instanceParam, instanceCast, levelParam, levelCast, loggingEventType);
 
-                _logDelegate = GetLogDelegate(loggerType, loggingEventType, instanceCast, instanceParam);
+                    _logDelegate = GetLogDelegate(_loggerType, loggingEventType, instanceCast, instanceParam);
 
-                _loggingEventPropertySetter = GetLoggingEventPropertySetter(loggingEventType);                
+                    _loggingEventPropertySetter = GetLoggingEventPropertySetter(loggingEventType);
+                }
+                catch { }
             }
 
             [SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "ILogger")]
+            [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
             internal Log4NetLogger(dynamic logger)
             {
-                _logger = logger.Logger;
+                try
+                {
+                    _logger = logger.Logger;
+
+                    if (_logEventLevelType == null)
+                        throw new InvalidOperationException("Type log4net.Core.Level was not found.");
```

These exceptions are just getting swallowed below?

---

Yes it is mostly about making people aware, that something is not working. Since throwing exceptions in the static-constructor is not recommended.